cmake/modules/BuildPlugin.cmake

```diff
@@ -56,11 +56,7 @@ MACRO(BUILD_PLUGIN PLUGIN_NAME)
 
 	ADD_LIBRARY(${PLUGIN_NAME} ${PLUGIN_LINK} ${PLUGIN_SOURCES} ${plugin_MOC_out} ${RCC_OUT})
 
-	TARGET_LINK_LIBRARIES(${PLUGIN_NAME} Qt5::Widgets Qt5::Xml)
-
-	IF(LMMS_BUILD_WIN32)
-		TARGET_LINK_LIBRARIES(${PLUGIN_NAME} lmms)
-	ENDIF(LMMS_BUILD_WIN32)
+	target_link_libraries("${PLUGIN_NAME}" lmms Qt5::Widgets Qt5::Xml)
```

Comment:
Kind of curious why the original code was only linking ``lmms`` for Windows builds (I'm guessing the plugins needed to see certain symbols on Windows)?

Reply:
That's right. If you reference an undefined symbol on Linux, the linker assumes it'll be available at runtime. This doesn't work on Windows - there's no global symbol table, so the linker needs to know which library each symbol will come from.

Reply:
I'm interested in this decision too.  I don't know enough about the symbol tables, but I'm curious what advantages (disadvantages) to toggle this for all systems.

I tried to dig around CMake docs to understand this better and just got more confused.  I did stumble upon this while I was searching though, which I found to be very interesting, perhaps unrelated to this PR https://stackoverflow.com/q/59838565/3196753.

Reply:
The immediate advantage is that dependencies become easier to manage, as any public dependencies of the core will be picked up automatically by plugins. It also allows plugins to consume symbols from the core in a consistent way across platforms (previously we only used this approach for Windows, and used some custom flags for macOS - the core dependency doesn't need to be specified on Linux as it only looks up the symbols at run time). I'm not aware of any disadvantages, and everything still seems to work, but that of course doesn't mean there aren't any.

Reply:
My first thought is that they stop being "plugins" from the sense that they're dependant on another library now.  Is this a misconception?

Reply:
It's not wrong, but I don't think it's relevant. The dependency was always present, just not as an explicit target-target dependency in CMake (except for Windows builds). IIRC, there have been discussions elsewhere about making the plugin API purer, so it doesn't require linking to the core, but that's not the goal here.

Reply:
> The dependency was always present, just not as an explicit target-target dependency in CMake (except for Windows builds)

I think of projects like https://github.com/pgiblock/lmms-lv2 are an interesting step in that direction.  If as you state, it's already a transient dependency, it's certainly off-topic for this discussion.